### DISCUSSION TITLE
Add ability to edit dag_run conf when re-running tasks

### DISCUSSION THREAD

**Discussion by veerendranathT** [apache/airflow#27860] [Ideas]
not able to edit conf section while re-run. Please find the below snap for the same.
![image](https://user-images.githubusercontent.com/46118531/118464772-d4ade480-b71e-11eb-8871-d831013d4da0.png)

**Comment by jakobkolb** [comment]
I have the same problem when triggering DAGs via CLI (which I use for integration testing during development). I would like to pass the dag a clean database via the `conf` to operate on but can't since the `conf` object is persisted between runs and cannot be reset afaik.
Passing a different json string via the `-c` flag is ignored after the first run.

**Comment by potiuk** [comment]
BTW. This is what is expected. Each manually triggered run is separate form another and you can always delete and recreate dag run if you want to re-run it during iterations. Re-run capability is foreseen for a different use case. Changing configuration for re-runs would be violating the idempotency property of the task, so we should not add it for "production" run . Deleting and creating run again while testing/integration is a very viable and "clean" solution. You will have to delete the run anyway eventually when you finish testing. And it's easy to automate.

**Reply by dakov** [reply]
I get your point, but imagine a scenario where you have some data processing in a kubernetes POD. It works fine all the time, but then you get a peak on your data.

Your regular run fails on insufficient resources. All that would be necessary to resolve the problem is to clear the existing dag run and update its params (assuming you have it configured to be passed into resources).

Since it is a regular run, adding a new manual run isn't really ideal. I assume you would end up with colliding logical date and you would have to trigger the whole DAG (maybe there are some costly operations - you could mark them as success, but then what's the point).

This is just and example, maybe you cannot pass resources to KubernetesPodOperator as params (I didn't check), but I hope my point is obvious

**Reply by potiuk** [reply]
Sure. If you want to propose  a PR including the UI modification to override params, I think it's a good PR to make. It's just not how it is implemented now - but If you can propose a change / PR implementing it,  I think it might be good idea.

**Comment by sdeangel** [comment]
Hi all, has that PR been implemented? I cannot find any documentation about it. 
Thanks
